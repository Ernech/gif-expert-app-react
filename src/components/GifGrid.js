import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
//import { getGIfs } from '../helpers/getGifs';
//import { GifGridItem } from './GifGridItem';



export const GifGrid =({category})=>{

//    const [images,setImages] = useState([]);


    const {data:images,loading} =useFetchGifs(category);
    console.log(loading);
   
 //   getGIfs();

    return(
        <>
        <h3>{category}</h3>
        {loading && <p>Loading</p>}
        <div className = "card-grid animate__animated animate__flash">
       
            {
                images.map(img =><GifGridItem key = {img.id} {...img}/>)
            }
       
        </div>
        </>
    )
}