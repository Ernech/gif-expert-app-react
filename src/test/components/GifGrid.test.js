import '@testing-library/jest-dom';
import React from 'react';
import {GifGrid} from '../../components/GifGrid';
import {shallow} from 'enzyme';
import {useFetchGifs} from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid/>', () => {
    const category = 'Music';
    test('Debe coincidir con el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        const wrapper = shallow(<GifGrid category = {category}/>)
         expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [{
            id:'ACB',
            url:'https://localhost/clualquiercosa.jpg',
            title:'Cualquier cosa'
        },
        {
            id:'ACB',
            url:'https://localhost/clualquiercosa.jpg',
            title:'Cualquier cosa'
        }];
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });
        const wrapper = shallow(<GifGrid category = {category}/>)
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

    });
});
