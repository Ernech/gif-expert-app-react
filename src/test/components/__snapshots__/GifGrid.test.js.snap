// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid/> Debe coincidir con el snapshot 1`] = `
<Fragment>
  <h3>
    Music
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__flash"
  />
</Fragment>
`;
